src/Agoda.Analyzers.Test/AgodaCustom/AG0026UnitTests.cs

```diff
@@ -0,0 +1,170 @@
+                           .AddMetadataReference(MetadataReference.CreateFromFile(typeof(IWebElement).Assembly.Location))
+                           .Documents
+                           .ToArray();
+
+            var diag = await GetSortedDiagnosticsFromDocumentsAsync(analyzers, documents, CancellationToken.None).ConfigureAwait(false);
+
+            var baseResult = CSharpDiagnostic(AG0026EnsureOnlyCssSelectorIsUsedToFindElements.DIAGNOSTIC_ID);
+            VerifyDiagnosticResults(diag, analyzers, new[]
+            {
+                baseResult.WithLocation(12, 82),
+                baseResult.WithLocation(14, 82),
+                baseResult.WithLocation(16, 82),
+                baseResult.WithLocation(18, 82),
+                baseResult.WithLocation(20, 82),
+                baseResult.WithLocation(22, 82),
+                baseResult.WithLocation(24, 82)
+            });
+        }
+
+        [Test]
+        public async Task AG0026_WhenUsedCssSelector_ThenNoWarning()
+        {
+            var testCode = @"
+            using System;
+            using OpenQA.Selenium;
+            using OpenQA.Selenium.Chrome;
+
+            namespace Selenium.Tests.Utils
+            {
+                public class FooterUtils
+                {
+                    private readonly string FooterSelector = ""[data-selenium=\""footer\""]"";
+
+                    public IWebElement Footer => new ChromeDriver().FindElement(By.CssSelector(FooterSelector));
+                }
+            }";
+
+            var analyzers = GetCSharpDiagnosticAnalyzers().ToImmutableArray();
+            var documents = CreateProject(new[] { testCode })
+                           .AddMetadataReference(MetadataReference.CreateFromFile(typeof(IWebElement).Assembly.Location))
+                           .Documents
+                           .ToArray();
+
+            var diag = await GetSortedDiagnosticsFromDocumentsAsync(analyzers, documents, CancellationToken.None).ConfigureAwait(false);
+
+            var baseResult = CSharpDiagnostic(AG0026EnsureOnlyCssSelectorIsUsedToFindElements.DIAGNOSTIC_ID);
+            VerifyDiagnosticResults(diag, analyzers, new DiagnosticResult[0]);
+        }
+
+        [Test]
+        public async Task AG0026_WhenMethodFromParent_ThenNoWarning()
+        {
+            var testCode = @"
+            using System;
+            using OpenQA.Selenium;
+            using OpenQA.Selenium.Chrome;
+
+            namespace Selenium.Tests.Utils
+            {
+                public class FooterUtils
+                {
+                    private readonly string FooterSelector1 = ""[data-selenium=\""footer1\""]"";
+
+                    private readonly string FooterSelector2 = ""[data-selenium=\""footer2\""]"";
+
+                    public bool IsEqualSelectors1 => By.Equals(FooterSelector1, FooterSelector2);
+
+                    public bool IsEqualSelectors2 => By.ReferenceEquals(FooterSelector1, FooterSelector2);
```

Comment:
Is there any value in this case? We're not testing for `GetHashCode` or `ToString`, both of which are also inherited from `Object`. I think you've proven your point with the case above.